apps/system/test/unit/notifications_test.js

```diff
@@ -646,6 +652,67 @@ suite('system/NotificationScreen >', function() {
     });
   });
 
+  suite('swiping to dismiss >', function() {
+    var notificationNode, notifClickedStub, contentNotificationEventStub;
+    var details = {
+      type: 'desktop-notification',
+      id: 'id-1',
+      title: '',
+      message: ''
+    };
+
+    setup(function() {
+      notificationNode = NotificationScreen.addNotification(details);
+
+      notifClickedStub = sinon.stub();
+      contentNotificationEventStub = sinon.stub();
+
+      window.addEventListener('notification-clicked', notifClickedStub);
+      window.addEventListener(
+        'mozContentNotificationEvent', contentNotificationEventStub);
+    });
+
+    function fakeEvt(x, y) {
+      return {
+        preventDefault: function() {},
+        touches: [{
+          target: notificationNode,
+          pageX: x,
+          pageY: y
+        }]
+      };
+    }
+
+    test('should disable scrolling during a swipe', function() {
+      var overflow = NotificationScreen.notificationsContainer.style.overflow;
+      assert.equal(overflow, '');
+
+      NotificationScreen.touchstart(fakeEvt(1, 1));
+      NotificationScreen.touchmove(fakeEvt(45, 1));
+      overflow = NotificationScreen.notificationsContainer.style.overflow;
+      assert.equal(overflow, 'hidden');
+
+      NotificationScreen.touchend(fakeEvt(45, 1));
+      overflow = NotificationScreen.notificationsContainer.style.overflow;
+      assert.equal(overflow, '');
+    });
+
+    test('should account for speed when dismissing', function() {
+      // Short but fast swipe
+      var close = this.sinon.stub(NotificationScreen, 'swipeCloseNotification');
+      NotificationScreen.touchstart(fakeEvt(1, 1));
+      this.sinon.clock.tick(10);
+      NotificationScreen.touchmove(fakeEvt(25, 1));
+      this.sinon.clock.tick(20);
+      NotificationScreen.touchmove(fakeEvt(45, 1));
+      this.sinon.clock.tick(30);
+      NotificationScreen.touchend(fakeEvt(45, 1));
+
+      sinon.assert.calledOnce(close);
+      sinon.assert.calledWith(close, 190);
```

Comment:
This is going to break if the animation speed changes I would suppose - could the unit test be altered to either read the animation time from the class, or to assert in a way that would allow small/medium changes to the animation time? (if it was a particularly fast swipe, I assume you could get the number pretty close to zero, then you could just test for < 50 or something like that?)